hardware_interface/include/joint_limits_interface/joint_limits_interface.hpp

```diff
@@ -45,33 +46,33 @@ class JointLimitHandle
 {
 public:
   JointLimitHandle()
-  : prev_pos_(std::numeric_limits<double>::quiet_NaN()),
-    prev_vel_(0.0),
-    jposh_("position"),
+  : jposh_("position"),
     jvelh_("velocity"),
-    jcmdh_("position_command")
+    jcmdh_("position_command"),
+    prev_pos_(std::numeric_limits<double>::quiet_NaN()),
+    prev_vel_(0.0)
   {}
 
   JointLimitHandle(
-    const hardware_interface::JointHandle & jposh,
-    const hardware_interface::JointHandle & jcmdh,
+    const hardware_interface::StateInterface & jposh,
+    hardware_interface::CommandInterface && jcmdh,
     const JointLimits & limits)
   : jposh_(jposh),
     jvelh_("velocity"),
-    jcmdh_(jcmdh),
+    jcmdh_(std::move(jcmdh)),
```

Comment:
If you do this here, will this interface be still accessible by controllers and hardware? Do you have a proposal how should `joint_limit_interface` be integrated into the current framework?